pennylane/interfaces/torch.py

```diff
@@ -505,12 +505,15 @@ def _res_to_torch(r, ctx):
     if isinstance(r, (list, tuple)):
         res = []
         for t in r:
-            if isinstance(t, dict):
+            if isinstance(t, dict) or isinstance(t, list) and all(isinstance(i, dict) for i in t):
+                # count result, single or broadcasted
                 res.append(t)
             else:
                 res.append(torch.as_tensor(t, device=ctx.torch_device))
         if isinstance(r, tuple):
             res = tuple(res)
+    elif isinstance(r, dict):
+        res = r

```

Comment:
unrelated to batching, but `counts` didn't work with the new return system at all because this wasn't here